aequilibrae/paths/linear_approximation.py

```diff
@@ -359,6 +363,22 @@ def execute(self):
                 # cost = c.fixed_cost / c.vot + self.congested_time #  now only once
                 cost = c.fixed_cost + self.congested_time
                 aggregate_link_costs(cost, c.graph.compact_cost, c.results.crosswalk)
+
+                if c._aon_results.save_path_file:
+                    # TODO (Jan 18/4/21): make base dir user configurable
+                    pth = os.environ.get(ENVIRON_VAR)
+                    path_base_dir = os.path.join(pth, "path_files", self.procedure_id, f"iter{self.iter}")
+                    c._aon_results.path_file_dir = os.path.join(path_base_dir, f"path_c{c.mode}_{c.__id__}")
+                    Path(c._aon_results.path_file_dir).mkdir(parents=True, exist_ok=True)
+                    if self.iter == 1:
```

Comment:
Is it here the place, or in the _execute_ of traffic assignment before the assignment actually starts?

Reply:
good question, I like having it in one place because the code is pretty monolithic already and hard to read, but the logic suggests doing this once before and then not checking for iteration numbers. I think I will try to make this more readable when thinking about the blob store.

Reply:
I've moved this into a little method and kept the if iter==1 statement in there, I like that it's self contained but let me know if you disagree